scripts/nb-tester/test-notebook.py

```diff
@@ -122,17 +122,17 @@ def extract_warnings(notebook: nbformat.NotebookNode) -> list[NotebookWarning]:
     return notebook_warnings
 
 
-def execute_notebook(path: Path, args: argparse.Namespace) -> bool:
+async def execute_notebook(path: Path, args: argparse.Namespace) -> bool:
     """
     Wrapper function for `_execute_notebook` to print status
     """
-    print(f"▶️  {path}", end="", flush=True)
+    print(f"▶️  {path}")
     possible_exceptions = (
         nbconvert.preprocessors.CellExecutionError,
         nbclient.exceptions.CellTimeoutError,
     )
     try:
-        nb = _execute_notebook(path, args)
+        nb = await _execute_notebook(path, args)
     except possible_exceptions as err:
         print("\r❌\n")
         print(err)
```

Comment:
These print statements assumed synchronicity, for async we need to remove the carriage return and make them make sense in isolation. Same for the warnings and success messages.

```suggestion
        print(f"❌ Problem in {path}\n", err)
```
Is this the printing problem you mentioned? Or are you not seeing output at all?

Reply:
I wasn't seeing anything print after the `await`, like no results at all. Only the starting line. I did not debug at all why that is and whether the notebook finishes running or not. Although I did get an exception raised with the save-jobs notebook 🤔

Reply:
FYI the problem was in line 167 which was passing the function return to `to_thread` rather than the function itself. See 92534d56a9f455273f03847e3bfbbcb3827174cb.